src/view/components/common/login-component/login-component.ts

```diff
@@ -0,0 +1,201 @@
+    encapsulation: ViewEncapsulation.Emulated,
+    moduleId: __moduleName, // REQUIRED to use relative path in styleUrls
+    providers: [LoginService],
+    selector: 'dot-login-component',
+    styleUrls: ['login-component.css'],
+    templateUrl: ['login-component.html'],
+})
+
+/**
+ * The login component allows the user to fill all
+ * the info required to log in the dotCMS angular backend
+ */
+export class LoginComponent {
+    @Input() my_account_login: string= '@dotcms.com';
+    @Input() password: string;
+    @Input() my_account_r_m: boolean= false;
+    @Input() forgotPasswordEmail: string;
+    @Output() toggleMain = new EventEmitter<boolean>();
+    languages: Array<string> = [];
+    @Input() language: string= 'en_US';
+    message: string= '';
+
+    // labels
+    loginLabel: string= '';
+    emailAddressLabel: string = ''
+    userIdOrEmailLabel: string= '';
+    passwordLabel: string= '';
+    rememberMeLabel: string= '';
+    forgotPasswordLabel: string= '';
+    loginButton: string= '';
+    forgotPasswordButton: string= '';
+    forgotPasswordConfirmationMessage: string= '';
+    cancelButton: string= '';
+    serverLabel: string = '';
+
+    _loginService: LoginService;
+    dotcmscompanyLogo: string= '';
+    dotcmsServerId: string= '';
+    dotcmslicenceLevel: string= '';
+    dotcmsVersion: string= '';
+    dotcmsBuildDateString: string= '';
+
+    private i18nMessages: Array<string> = [ 'Login', 'email-address', 'user-id', 'password', 'remember-me', 'sign-in', 'forgot-password', 'get-new-password', 'cancel', 'an-email-with-instructions-will-be-sent','Server'];
+
+    constructor( @Inject('menuItems') private menuItems: any[], private _service: LoginService) {
+        this._loginService = _service;
+        this.updateScreenBackground();
+    }
+
+    /**
+     *  Executes the logIn service
+     */
+    logInUser(): void {
+
+        this._loginService.logInUser(this.my_account_login, this.password, this.my_account_r_m, this.language).subscribe((result: any) => {
+            if (result.errors.length > 0) {
+                this.message = result.errors[0].message;
+            } else {
+                this.message = '';
+                this.toggleMain.emit(false);
+                // this window.location.reload should be removed once the menu and router injection update issue is fixed
+                window.location.reload();
+            }
+        }, (error) => {
+            if (error.response.status === 400 || error.response.status === 401) {
+                this.message = this.getErrorMessage(error);
+            } else {
+                console.log(error);
+            }
+        });
+
+    }
+
+    /**
+     * Executes the recover password service
+     */
+    recoverPassword(): void {
+        if (confirm(this.forgotPasswordConfirmationMessage)) {
+            document.getElementById('forgotPassword').className = 'forgotPasswordBox hideBox';
```

Comment:
This is definitely a NO-NO... in Angular you can set classNames in the html template base on the result of expressions: 

```
<div [class.extra-sparkle]="isDelightful"> // Binds the presence of the CSS class extra-sparkle on the element to the truthiness of the expression isDelightful.
```